distributed/core.py

```diff
@@ -132,12 +132,12 @@ class Server(TCPServer):
     default_port = 0
 
     def __init__(self, handlers, max_buffer_size=MAX_BUFFER_SIZE,
-            connection_limit=512, **kwargs):
+            connection_limit=512, ssl_context_creator= create_ssl_context, **kwargs):
         self.handlers = assoc(handlers, 'identity', self.identity)
         self.id = str(uuid.uuid1())
         self._port = None
         self.rpc = ConnectionPool(limit=connection_limit)
-        super(Server, self).__init__(max_buffer_size=max_buffer_size, **kwargs)
+        super(Server, self).__init__(max_buffer_size=max_buffer_size, ssl_options=ssl_context_creator(), **kwargs)
```

Comment:
Yeah, so I think at the moment I'm mostly confused because I don't understand the pain you're currently trying to solve.  If I were to solve this problem (that I admittedly know little about) I would probably check for ssl certificates whereever I called Scheduler/Worker and then send them in at that point.  My guess is that this would be at the CLI level.